src/mesh/config/MeshConfiguration.cpp

```diff
@@ -86,7 +86,7 @@ void MeshConfiguration::xmlTagCallback(
     bool        found = false;
     for (const DataConfiguration::ConfiguredData &data : _dataConfig->data()) {
       if (data.name == name) {
-        _meshes.back()->createData(data.name, data.dimensions, _dataIDManager.getFreeID(), data.hasGradient);
+        _meshes.back()->createData(data.name, data.dimensions, _dataIDManager.getFreeID());
```

Comment:
I quickly skimmed over the places where `_meshes` or `meshes()` are used and overall I have the impression that at most places replacing the vector with a dict would help. One could also introduce some more high-level functions in `MeshConfiguration` to avoid call-chains like `meshConfig->meshes().at(0)->createVertex(Eigen::Vector3d(1.0, 1.0, 1.0));` in the client code.

Reply:
What is a dict? Do you mean map? I agree that there are several places, which are just confusing. If we can boil down 'problem classes' we should open an issue and discuss how to adapt things. I would prefer to avoid adding such changes in this PR.